File: app/src/main/java/com/androidexamples/app/domain/ParserJsonExemple.java
package com.androidexamples.app.domain;

import android.content.Context;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by marcos on 15/01/2017.
 */

public class ParserJsonExemple {

    private List<Example> listExamples;

    public List<Example> getExamples(Context context, String json) {
        List<Example> listExamples = new ArrayList<>();

        try {
            JSONArray jsonArray = new JSONArray(json);

            for(int i = 0; i < jsonArray.length(); i++) {
                JSONObject jsonObject = jsonArray.getJSONObject(i);
                Example example = new Example();
                example.setId(jsonObject.optInt("id"));
                example.setName(jsonObject.optString("name"));
                example.setDeleted(jsonObject.optBoolean("deleted"));
                listExamples.add(example);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return listExamples;
    }
}

class Example {
    private int id;
    private String name;
    private boolean deleted;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isDeleted() {
        return deleted;
    }

    public void setDeleted(boolean deleted) {
        this.deleted = deleted;
    }
}

File: app/src/main/java/com/androidexamples/app/domain/ParserJsonExemple.kt
package com.androidexamples.app.domain

import android.content.Context

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

/**
 * Created by marcos on 15/01/2017.
 */

class ParserJsonExemple {

    private val listExamples: List<Example>? = null

    fun getExamples(context: Context, json: String): List<Example> {
        val listExamples = ArrayList<Example>()

        try {
            val jsonArray = JSONArray(json)

            for (i in 0 until jsonArray.length()) {
                val jsonObject = jsonArray.getJSONObject(i)
                val example = Example()
                example.id = jsonObject.optInt("id")
                example.name = jsonObject.optString("name")
                example.isDeleted = jsonObject.optBoolean("deleted")
                listExamples.add(example)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return listExamples
    }
}

class Example {
    var id: Int = 0
    var name: String? = null
    var isDeleted: Boolean = false
}
